From a5a7c3ad5ce91cfa7e663b37cb546f157b37ef91 Mon Sep 17 00:00:00 2001
From: John Sullivan <jack.t.sullivan@oracle.com>
Date: Thu, 2 Sep 2021 10:50:56 -0400
Subject: [PATCH 2/2] added nulling out for defaultValue in
 FieldResponseProcesser

---
 .../columnar/processors/response/FieldResponseProcessor.java     | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Data/src/main/java/org/tribuo/data/columnar/processors/response/FieldResponseProcessor.java b/Data/src/main/java/org/tribuo/data/columnar/processors/response/FieldResponseProcessor.java
index f7d9dcf..397e9fa 100644
--- a/Data/src/main/java/org/tribuo/data/columnar/processors/response/FieldResponseProcessor.java
+++ b/Data/src/main/java/org/tribuo/data/columnar/processors/response/FieldResponseProcessor.java
@@ -67,6 +67,7 @@ public class FieldResponseProcessor<T extends Output<T>> implements ResponseProc
         } else if (fieldNames != null) {
             if(defaultValue != null) {
                 defaultValues = defaultValues == null ? Collections.nCopies(fieldNames.size(), defaultValue) : defaultValues;
+                defaultValue = null;
             } else {
                 throw new PropertyException(configName, "defaultValue, defaultValues", "one of defaultValue or defaultValues must be populated");
             }
-- 
2.29.3

